autogen/AudioSource.m

```diff
@@ -0,0 +1,78 @@
+% AudioSource 
+%
+% This class represents an audio source for use with Autogen.
+%
+% Copyright 2019 Audio Logic
+%
+% THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+% INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+% PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+% FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+% ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+%
+% Dylan Wickham
+% Audio Logic
+% 985 Technology Blvd
+% Bozeman, MT 59718
+% openspeech@flatearthinc.com
+classdef AudioSource
+    %AudioSource Represents an audio source for use with Autogen
+    properties
+        audio
+        sampleRateHz
+    end
+    
+    properties(SetAccess = private)
+        nSamples
+        nChannel
+        duration
+    end
+    
+    methods
+        function obj = AudioSource(audio, sampleRateHz)
+        %AudioSource Initializes AudioSource with m x n audio input and
+        % sample rate in hz, where m is the the number of samples and n is
+        % the number of channels.
```

Comment:
Do we want to have support for taking an audio file path as input?

Reply:
We could potentially. That is currently what `sm_init_test_signals` is handling, but moving that into something like AudioSource.fromFile() is an option.

Reply:
`sm_init_test_signals` still has configuration in the case statement, which isn't particularly useful anymore since it's moved into the autogen directory. 

Is the fastsim functionality still being used, or are you running simulations at the sample rate? We are moving towards the latter elsewhere, so maybe we can get rid of `sm_init_test_signals` right now. What are your thoughts?

Reply:
Indeed `sm_init_test_signals` could have the switch statement removed as it is currently only intended to process an the given audio file into an AudioSource. The fastsim flag is implemented (it is still useful for getting fast simulations). Given that the fastsim functionality is used, the actual signature if the `sm_init_test_signals` functionality is moved would be more like `AudioSource.fromFile(filepath, fastsim_flag, fastsim_Nsamples)` or `AudioSource.fromFile(filepath, nSamples)` and then set `nSamples` to -1 when the full file is to be used.

Reply:
To further clarify, adjusting the number of samples for a simulation will for obvious reasons have a large effect on the time the simulation takes. However after a bug a fix is taken care of, adjusting `fastsim_Fs_system_N` doesn't do anything and could be removed, because I set the "system clock" to be the sampling rate times the number of channels currently, as anything more is not really useful during simulation. (The system clock doesn't actually matter in simulation, just the time step which is what is actually being set)

Given that, a `fastsim_flag` doesn't seem that useful just for enabling `fastsim_Nsamples` we could get rid of `fastsim_flag` and rename `fastsim_Nsamples` to just `nSamples` with -1 being the full audio file.

Reply:
Updated to this

Reply:
Done